dev/NavigationView/NavigationView.cpp

```diff
@@ -1538,6 +1548,11 @@ void NavigationView::RaiseSelectionChangedEvent(winrt::IInspectable const& nextI
 // If nextItem is selectionsuppressed, we should undo the selection. We didn't undo it OnSelectionChange because we want change by API has the same undo logic.
 void NavigationView::ChangeSelection(const winrt::IInspectable& prevItem, const winrt::IInspectable& nextItem)
 {
+    // Selection changed event was requested to be ignored, so let's do that
+    if (m_shouldIgnoreNextSelectionChange) {
```

Comment:
Using the `m_shouldIgnoreNextSelectionChange` flag might cause issues.
This is because this flag is currently being used to block selection in for the selectionmodel in the `OnSelectionModelSelectionChanged` function.
For an example, take a look in the `void NavigationView::UpdateSelectionForMenuItems()` function.

Reply:
Yeah that may create some problems. Would introducing a new flag like `m_shouldIgnoreNextSelectionChangeSettingsItem` be an acceptable fix for this? 

The problem is that by the time `ChangeSelection` gets called, we can not accurately verify if we are switching display mode and settings item needs to be switched or if there was another reason the selection changed.

Reply:
Yeah, that sounds like a reasonable move to me.